### PR TITLE
[Java] Fix deselecting options in <select>

### PR DESCRIPTION
**Pull request by glibas**
1. Fix inconsistency between java doc and implementation:
   deselectByVisibleText, deselectByIndex and deselectByValue
   did not throw NoSuchElementException, while they should have to.
2. Remove check !isMultiple() while selecting/deselecting by index,
   since no need as index is unique.
3. Add appropriate tests to SelectElementTest
4. Include SelectElementTest in LargeTests suite, since it was
   not a part of any.

### CODE REVIEW

File: java/client/src/org/openqa/selenium/support/ui/Select.java [hunk 1]
```diff
@@ -283,15 +282,31 @@ public void deselectByVisibleText(String text) {
...
+   * Select or deselect specified option
+   *
+   * @param option
+   *          The option which state needs to be changed
+   * @param select
+   *          Indicates whether the option needs to be selected (true) or
+   *          deselected (false)
+   */
+  private void setSelected(WebElement option, boolean select) {
+    if ((!option.isSelected() && select) || (option.isSelected() && !select)) {
```
Comment (lukeis) [inline java/client/src/org/openqa/selenium/support/ui/Select.java]: it's not efficient to make the remote call to isSelected multiple times here. Please either revert the change to this method or store the isSelected value in a local variable.

Comment (glib-briia) [inline java/client/src/org/openqa/selenium/support/ui/Select.java]: Hi @lukeis,

Thank you for your comment. It is indeed. Updated in commit f326050e1d611da34dbce9748de6d627efd46f3f.

File: java/client/src/org/openqa/selenium/support/ui/Select.java [hunk 2]
```diff
@@ -245,11 +237,16 @@ public void deselectAll() {
   public void deselectByValue(String value) {
     List<WebElement> options = element.findElements(By.xpath(
         ".//option[@value = " + Quotes.escape(value) + "]"));
-
+    boolean matched = false;
     for (WebElement option : options) {
-      if (option.isSelected()) {
-        option.click();
+      setSelected(option, false);
+      if (!isMultiple()) {
```
Comment (lukeis) [inline java/client/src/org/openqa/selenium/support/ui/Select.java]: deselecting a non-multiple select doesn't make sense. What would it even mean to deselect something in a normal select? A user can't do it.

Rather than adding this check here (same goes for deselectByVisibleText), we could check at the start of the method and throw an exception if it is not multiple.

**Comment by lukeis** [comment]
also can you confirm you've signed the CLA? http://goo.gl/qC50R

**Comment by glib-briia** [comment]
Yes, you are right, thank you. Updated in commit bf60cb9909e8ddb84a8ece88e558c5a6c04a108d. 
Appropriate tests added. 

_CLI is signed._

**Comment by lukeis** [comment]
Thanks!

merged as 5883af125a72d901b7fbe79664cce9fa4d7983a8
